cadence/contracts/ISPOManager.cdc

```diff
@@ -353,9 +356,15 @@ pub contract ISPOManager {
         }
 
         pub fun withdrawAdminFlowRewards(): @FungibleToken.Vault {
+            if self.stakingRewardsVault == nil {
+                self.stakingRewardsVault <-! FlowToken.createEmptyVault()
+            }
             let stakingRewardsVaultRef: &FungibleToken.Vault? = &self.stakingRewardsVault as &FungibleToken.Vault?
             for key in self.delegators.keys {
-                let delegatorRecordRef: &ISPOManager.DelegatorRecord = self.borrowDelegatorRecord(delegatorId: key)
+                let delegatorRecordRef: &ISPOManager.DelegatorRecord = self.borrowDelegatorRecord(delegatorId: key)!
+                if delegatorRecordRef == nil {
```

Comment:
Is this logic ever triggered? Seems to me that we do not destroy delegator records so for each key, there must be a delegatorRecord, eg. changing the `borrowDelegatorRecord` to return optional does not seem to be needed.

Reply:
user may withdraw their delegation and in this case this ref becomes nil, this also creates the edge-case that I'm treating above